action/protocol/poll/nativestaking.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package poll
+
+import (
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/iotexproject/iotex-election/types"
+)
+
+var dummyCaller, _ = address.FromString("io1emxf8zzqckhgjde6dqd97ts0y3q496gm3fdrl6")
+
+type (
+	// NativeStaking represents native staking struct
+	NativeStaking struct {
+		cm       protocol.ChainManager
+		contract string
+		abi      abi.ABI
+	}
+
+	pygg struct {
+		count           *big.Int
+		indexes         []*big.Int
+		stakeStartTimes []*big.Int
+		stakeDurations  []*big.Int
+		decays          []bool
+		stakedAmounts   []*big.Int
+		canNames        [][12]byte
+		owners          []common.Address
+	}
+)
+
+// NewNativeStaking creates a NativeStaking instance
+func NewNativeStaking(cm protocol.ChainManager, staking string) (*NativeStaking, error) {
+	abi, err := abi.JSON(strings.NewReader(NsAbi))
+	if err != nil {
+		return nil, err
+	}
+	return &NativeStaking{cm, staking, abi}, nil
+}
+
+// Votes returns the votes on height
+func (ns *NativeStaking) Votes(height uint64) ([]*types.Vote, error) {
```

Comment:
I recommend add height later if we really need it

Reply:
yes, we'll need to pass in height now